engine/verification/assigner/engine.go

```diff
@@ -182,51 +178,35 @@ func (e *Engine) processFinalizedBlock(ctx context.Context, block *flow.Block) {
 	assignedChunksCount := uint64(0)
 	processedChunksCount := uint64(0)
 
-	log := e.log.With().
+	lg := e.log.With().
 		Hex("block_id", logging.ID(blockID)).
-		Uint64("block_height", block.Header.Height).
-		Int("receipt_num", len(block.Payload.Receipts)).Logger()
-
-	log.Debug().Msg("new finalized block arrived")
-
-	resultsById := block.Payload.ResultsById()
-
-	// performs chunk assigment on each receipt and pushes the assigned chunks to the
-	// chunks queue.
-	for _, meta := range block.Payload.Receipts {
-
-		// This is a TEMPORARY SHORTCUT; mature solution will be implemented in
-		// https://github.com/dapperlabs/flow-go/issues/5397
-		result, isIncorporated := resultsById[meta.ResultID]
-		if !isIncorporated {
-			// Per protocol definition, a result is only incorporated once in each fork,
-			// specifically in the first block that contains an execution receipt committing to the result.
-			// Hence, if we find a receipt whose result is _not_ incorporated in this block, the result
-			// was incorporated in an ancestor block. Consequently, we have already verified the result
-			// previously and don't need to do it again.
-			continue
+		Uint64("block_height", block.Header.Height).Logger()
+	lg.Debug().Int("result_num", len(block.Payload.Results)).Msg("new finalized block arrived")
+
+	// determine chunk assigment on each result and pushes the assigned chunks to the chunks queue.
+	receiptsGroupedByResultID := block.Payload.Receipts.GroupByResultID() // for logging purposes
+	for _, result := range block.Payload.Results {
+		resultID := result.ID()
+
+		// log receipts committing to result
+		resultLog := lg.With().Hex("incorporated_result_id", logging.ID(resultID)).Logger()
```

Comment:
is it an incorporated result?

Reply:
good point: the data structure is a `Result` (which is incorporated in a block), but it is not an `IncorporatedResult` (which has a different id). 

updated in https://github.com/onflow/flow-go/pull/546/commits/282e833f6c0e7903846076da949f7dbf625f6e9e